File: update_navs.py
import os
import re

# Read index.html to get the nav container content
with open('index.html', 'r', encoding='utf-8') as f:
    index_html = f.read()

# Extract the inner HTML of the nav
nav_match = re.search(r'<nav class="navbar[^>]*>\s*(<div class="container">.*?</button>\s*</div>)\s*</nav>', index_html, re.DOTALL)
if not nav_match:
    print("Could not find nav in index.html")
    exit(1)

nav_inner = nav_match.group(1)

# Now fix the paths in nav_inner for the pages directory
# 1. href="pages/xxx.html" -> href="xxx.html"
nav_inner_pages = re.sub(r'href="pages/([^"]+)"', r'href="\1"', nav_inner)
# 2. href="index.html" -> href="../index.html"
nav_inner_pages = re.sub(r'href="index\.html"', r'href="../index.html"', nav_inner_pages)
# 3. src="assets/..." -> src="../assets/..."
nav_inner_pages = re.sub(r'src="assets/', r'src="../assets/', nav_inner_pages)

# Files to ignore
ignore_files = ['login.html', 'dashboard.html', 'signup.html']

pages_dir = 'pages'
for filename in os.listdir(pages_dir):
    if filename.endswith('.html') and filename not in ignore_files:
        filepath = os.path.join(pages_dir, filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
            
        # Replace the inner nav
        new_content = re.sub(r'(<nav class="navbar[^>]*>)\s*<div class="container">.*?</button>\s*</div>\s*(</nav>)', 
                             lambda m: f"{m.group(1)}\n      {nav_inner_pages}\n  {m.group(2)}", 
                             content, flags=re.DOTALL)
                             
        # Fix the active class
        # First, remove active from all nav-links
        new_content = re.sub(r'class="nav-link active"', 'class="nav-link"', new_content)
        # Then, add active to the current page
        page_name = filename
        # Special cases or direct matches
        new_content = re.sub(f'href="{page_name}" class="nav-link"', f'href="{page_name}" class="nav-link active"', new_content)
        
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated {filename}")

print("Done.")

File: update_navs.js
const fs = require('fs');
const path = require('path');

const indexHtml = fs.readFileSync('index.html', 'utf8');
const navMatch = indexHtml.match(/<nav class="navbar[^>]*>\s*(<div class="container">[\s\S]*?<\/button>\s*<\/div>)\s*<\/nav>/);

if (!navMatch) {
    console.error("Could not find nav in index.html");
    process.exit(1);
}

let navInner = navMatch[1];

// 1. href="pages/xxx.html" -> href="xxx.html"
navInner = navInner.replace(/href="pages\/([^"]+)"/g, 'href="$1"');
// 2. href="index.html" -> href="../index.html"
navInner = navInner.replace(/href="index\.html"/g, 'href="../index.html"');
// 3. src="assets/..." -> src="../assets/..."
navInner = navInner.replace(/src="assets\//g, 'src="../assets/');

const ignoreFiles = ['login.html', 'dashboard.html', 'signup.html'];
const pagesDir = 'pages';

const files = fs.readdirSync(pagesDir);

for (const filename of files) {
    if (filename.endsWith('.html') && !ignoreFiles.includes(filename)) {
        const filepath = path.join(pagesDir, filename);
        let content = fs.readFileSync(filepath, 'utf8');
        
        content = content.replace(/(<nav class="navbar[^>]*>)\s*<div class="container">[\s\S]*?<\/button>\s*<\/div>\s*(<\/nav>)/, 
                                  `$1\n    ${navInner}\n  $2`);
        
        // Remove active class from all links
        content = content.replace(/class="nav-link active"/g, 'class="nav-link"');
        
        // Add active class to current page
        // Special case: home2.html uses regex
        const pageName = filename;
        const regex = new RegExp(`href="${pageName}" class="nav-link"`, 'g');
        content = content.replace(regex, `href="${pageName}" class="nav-link active"`);
        
        fs.writeFileSync(filepath, content, 'utf8');
        console.log(`Updated ${filename}`);
    }
}

// Add version busting to components.css in all pages to fix the hamburger menu bug there too!
for (const filename of files) {
    if (filename.endsWith('.html')) {
        const filepath = path.join(pagesDir, filename);
        let content = fs.readFileSync(filepath, 'utf8');
        content = content.replace(/assets\/css\/components\.css(?![\?])/g, 'assets/css/components.css?v=3');
        content = content.replace(/assets\/css\/variables\.css(?![\?])/g, 'assets/css/variables.css?v=3');
        content = content.replace(/assets\/css\/global\.css(?![\?])/g, 'assets/css/global.css?v=3');
        fs.writeFileSync(filepath, content, 'utf8');
    }
}
console.log("CSS versioning updated for all pages.");

console.log("Done.");
